### PR TITLE
Strict filters

### PR DESCRIPTION
**Pull request by EvanBarbour3**
## Bug Fixes
- Added strict filters configuration to stop all filters performing a `%like%`

[Delivers [#120571545](https://www.pivotaltracker.com/story/show/120571545)]

### CODE REVIEW

File: app/Http/Traits/FilterTrait.php [hunk 1]
```diff
@@ -26,17 +26,24 @@
         'ext_finance_net_settlement'
     ];
 
+    abstract protected function getStrictFiltersConfiguration();
```
Comment (wnowicki) [inline app/Http/Traits/FilterTrait.php]: - DocBlock is missing
- Idea was to deliver configuration of filters like:
```php
[
    'id' => self::FILTER_STRICT,
    'ext_order_amount' => self::FILTER_FINANCE,
    'ext_received' => self::FILTER_DATE_RANGE,
]
```
 - if not defined on a list treated as default behaviour

**Comment by EvanBarbour3** [comment]
@wnowicki

File: app/Http/Controllers/ApplicationsController.php [hunk 2]
```diff
@@ -345,4 +345,23 @@ private function renderConfirmationScreen($action, $id, $installation)
...
+    {
+        return [
+            'ext_order_amount' => Controller::FILTER_FINANCE,
+            'ext_finance_loan_amount' => Controller::FILTER_FINANCE,
+            'ext_finance_deposit' => Controller::FILTER_FINANCE,
+            'ext_finance_subsidy' => Controller::FILTER_FINANCE,
+            'ext_finance_net_settlement' => Controller::FILTER_FINANCE,
+        ];
+    }
+
```
Comment (wnowicki) [inline app/Http/Controllers/ApplicationsController.php]: - CS

Comment (wnowicki) [inline app/Http/Controllers/ApplicationsController.php]: - CS

**Pull request merged**